src/platform/packages/shared/kbn-esql-validation-autocomplete/src/autocomplete/__tests__/autocomplete.command.stats.test.ts

```diff
@@ -272,24 +362,25 @@ describe('autocomplete.suggest', () => {
           'var0 = ',
           getDateHistogramCompletionItem(),
           ...fields,
-          ...getFunctionSignaturesByReturnType('eval', 'any', { scalar: true }),
+          ...getFunctionSignaturesByReturnType(Location.EVAL, 'any', { scalar: true }),
           ...allGroupingFunctions,
         ]);
       });
 
       test('on space before expression right hand side operand', async () => {
-        const { assertSuggestions } = await setup();
-
-        await assertSuggestions('from a | stats avg(b) by integerField % /', [
-          ...getFieldNamesByType('integer'),
-          ...getFieldNamesByType('double'),
-          ...getFieldNamesByType('long'),
-          ...getFunctionSignaturesByReturnType('eval', ['integer', 'double', 'long'], {
-            scalar: true,
-          }),
-          // categorize is not compatible here
-          ...allGroupingFunctions.filter((f) => !f.text.includes('CATEGORIZE')),
-        ]);
+        /**
+         * @todo re-enable when expression autocomplete is officially supported for STATS ... BY
+         */
+        // await assertSuggestions('from a | stats avg(b) by integerField % /', [
```

Comment:
Why these tests are commented out? Our changes should not affect them

Reply:
When we detected an operator in STATS we were sending the code to `getFunctionArgsSuggestions` which worked in this case. So it was kind of a half-hearted attempt to support expressions in the `STATS <here>`.

I removed that logic from `getAstContext` because it was sending us to the wrong place in the `STATS ... WHERE <here>`. 

At the time, I was thinking I would add first-class expressions support for this case too, but that is for later in https://github.com/elastic/kibana/issues/216492.

I should be able to adjust the `getAstContext` logic to restore the original behavior without ruining the work done for `STATS... WHERE`.

Reply:
Restored in https://github.com/elastic/kibana/pull/216379/commits/40ada593d203411a3f4bda0996040e2c533fd175